aiida_quantumespresso/calculations/neb.py

```diff
@@ -5,438 +5,316 @@
-        settings = inputdict.pop(self.get_linkname('settings'), None)
-        if settings is None:
-            settings_dict = {}
+        # Convert settings dictionary to have uppercase keys, or create an empty one if none was given.
+        if 'settings' in self.inputs:
+            settings_dict = _uppercase_dict(self.inputs.settings.get_dict(), dict_name='settings')
         else:
-            if not isinstance(settings,  Dict):
-                raise InputValidationError('settings, if specified, must be of '
-                                           'type Dict')
-            # Settings converted to uppercase
-            settings_dict = _uppercase_dict(settings.get_dict(),
-                                            dict_name='settings')
-
-        pseudos = {}
-        # I create here a dictionary that associates each kind name to a pseudo
-        for link in inputdict.keys():
-            if link.startswith(self._get_linkname_pseudo_prefix()):
-                kindstring = link[len(self._get_linkname_pseudo_prefix()):]
-                kinds = kindstring.split('_')
-                the_pseudo = inputdict.pop(link)
-                if not isinstance(the_pseudo, UpfData):
-                    raise InputValidationError('Pseudo for kind(s) {} is not of '
-                                               'type UpfData'.format(','.join(kinds)))
-                for kind in kinds:
-                    if kind in pseudos:
-                        raise InputValidationError('Pseudo for kind {} passed '
-                                                   'more than one time'.format(kind))
-                    pseudos[kind] = the_pseudo
-
-        parent_calc_folder = inputdict.pop(self.get_linkname('parent_folder'), None)
-        if parent_calc_folder is not None:
-            if not isinstance(parent_calc_folder, RemoteData):
-                raise InputValidationError('parent_calc_folder, if specified, '
-                                           'must be of type RemoteData')
-
-        vdw_table = inputdict.pop(self.get_linkname('vdw_table'), None)
-        if vdw_table is not None:
-            if not isinstance(vdw_table, SinglefileData):
-                raise InputValidationError('vdw_table, if specified, '
-                                           'must be of type SinglefileData')
-
-        # Here, there should be no more parameters...
-        if inputdict:
-            raise InputValidationError('The following input data nodes are '
-                                       'unrecognized: {}'.format(list(inputdict.keys())))
+            settings_dict = {}
+
+        parent_calc_folder = self.inputs.get('parent_folder', None)
+        vdw_table = self.inputs.get('pw.vdw_table', None)
 
         # Check that the first and last image have the same cell
-        if abs(np.array(first_structure.cell)-
-               np.array(last_structure.cell)).max() > 1.e-4:
+        if abs(np.array(self.inputs.first_structure.cell)-
```

Comment:
better to dereference `first_structure` and `last_structures` once and assign to variables, because each time you call it, it will perform database operation

Reply:
Good to know, I'll do it.
Is this the case for all types of nodes? And do they have all the same cost? The idiom `self.inputs.whatever` is clearer in my opinion, so I would try to use it when the cost is small.